include/swift/SIL/SILInstruction.h

```diff
@@ -7978,6 +7982,131 @@ inline DestructureTupleInst *DestructureTupleResult::getParent() {
   return cast<DestructureTupleInst>(Parent);
 }
 
+/// SWIFT_ENABLE_TENSORFLOW
+/// A result for the graph_op instruction. See documentation for
+/// graph_op for more information.
+class GraphOperationResult final : public MultipleValueInstructionResult {
+public:
+  GraphOperationResult(unsigned Index, SILType Type,
+                       ValueOwnershipKind OwnershipKind)
+  : MultipleValueInstructionResult(ValueKind::GraphOperationResult, Index,
+                                   Type, OwnershipKind) {}
+
+  static bool classof(const SILNode *N) {
+    return N->getKind() == SILNodeKind::GraphOperationResult;
+  }
+
+  GraphOperationInst *getParent();
+  const GraphOperationInst *getParent() const {
+    return const_cast<GraphOperationResult *>(this)->getParent();
+  }
+};
+
+/// SWIFT_ENABLE_TENSORFLOW
+/// An graph operation attribute.
+/// Currently, attributes have a name and a constant value.
+struct GraphOperationAttribute {
+  Identifier name;
+  SymbolicValue value;
+};
+
+/// SWIFT_ENABLE_TENSORFLOW
+/// A graph operation. This instruction will be extracted to a graph program
+/// via graph program extraction passes.
+class GraphOperationInst final
+  : public InstructionBase<
+               SILInstructionKind::GraphOperationInst,
+               MultipleValueInstruction>,
+    public MultipleValueInstructionTrailingObjects<
+               GraphOperationInst, GraphOperationResult,
+               InitialTrailingObjects<>,
+               FinalTrailingObjects<Operand, GraphOperationAttribute>> {
+  friend TrailingObjects;
+
+  /// The name of the graph operation.
+  Identifier Name;
+  /// The number of operands.
+  unsigned NumOperands;
+  /// The number of attributes.
+  unsigned NumAttributes;
+
+  GraphOperationInst(SILModule &M, SILDebugLocation loc, Identifier name,
+                     ArrayRef<SILValue> arguments,
+                     ArrayRef<GraphOperationAttribute> attrs,
+                     ArrayRef<SILType> resultTypes,
+                     ArrayRef<ValueOwnershipKind> resultOwnerships) :
+    InstructionBase(loc),
+    MultipleValueInstructionTrailingObjects(this, resultTypes,
+                                            resultOwnerships),
+    Name(name), NumOperands(arguments.size()), NumAttributes(attrs.size()) {
+      auto allOperands = getAllOperands();
+      for (unsigned i : indices(arguments))
+        new (&allOperands[i]) Operand(this, arguments[i]);
+      std::copy(attrs.begin(), attrs.end(), getAttributes().data());
```

Comment:
I think this should use uninitialized_copy ?

Reply:
Because the destination is uninitialized memory

Reply:
Done.